_re2.cc

```diff
@@ -403,14 +415,33 @@ _do_search(RegexpObject2* self, PyObject* args, PyObject* kwds, RE2::Anchor anch
 
   // Using O! instead of s# here, because we want to stash the original
   // PyObject* in the match object on a successful match.
-  if (!PyArg_ParseTupleAndKeywords(args, kwds, "O!|ll", (char**)kwlist,
-        &PyString_Type, &string,
+  if (!PyArg_ParseTupleAndKeywords(args, kwds, "O|ll", (char**)kwlist,
+        &string,
         &pos, &endpos)) {
     return NULL;
   }
 
-  const char* subject = PyString_AS_STRING(string);
-  Py_ssize_t slen = PyString_GET_SIZE(string);
+  const char *subject;
+  Py_ssize_t slen;
+#if PY_MAJOR_VERSION >= 3
+  if (PyUnicode_Check(string)) {
+    subject = PyUnicode_AsUTF8AndSize(string, &slen);
+  } else if (PyBytes_Check(string)) {
+    subject = PyBytes_AS_STRING(string);
+    slen = PyBytes_GET_SIZE(string);
+  } else {
+    Py_DECREF(string);
```

Comment:
Why is this a decref?  https://docs.python.org/2/c-api/arg.html says that O doesn't increase refcount.

Reply:
Good catch, these were left over from some previous refactors on my side. Removed both decrefs as they're not needed.